packages/block-editor/src/components/link-control/search-create-button.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { cleanForSlug } from '@wordpress/url';
+import {
+	Button,
+	Icon,
+} from '@wordpress/components';
+
+export const LinkControlSearchCreate = ( { searchTerm = '', onClick, itemProps } ) => {
+	if ( ! searchTerm.trim() ) {
+		return null;
+	}
+
+	return (
+		<Button
+			role="option"
+			{ ...itemProps }
+			onClick={ onClick }
+			className={ classnames( 'block-editor-link-control__search-create block-editor-link-control__search-item', {
+			} ) }
+		>
+
+			<Icon className="block-editor-link-control__search-item-icon" icon="insert" />
+
+			<span className="block-editor-link-control__search-item-header">
+				<span className="block-editor-link-control__search-item-title">
+					{ sprintf( __( 'Create new Page: %s' ), searchTerm ) }
+				</span>
+				<span className="block-editor-link-control__search-item-info">
+					/{ cleanForSlug( searchTerm ) }/
```

Comment:
We can remove this `slug` in favour of what @shaunandrews 's mockups show. As we said, try to display a slug before Page is created is misleading.